File: backend/src/main/java/at/tourplanner/tour_planner/api/dto/tourlog/TourLogResponse.java
package at.tourplanner.tour_planner.api.dto.tourlog;

import at.tourplanner.tour_planner.features.tourlog.TourLog;

import java.math.BigDecimal;
import java.time.LocalDateTime;

public record TourLogResponse(
        Long tourLogId,
        Long tourId,
        String comment,
        Integer difficulty,
        Integer rating,
        Long totalTimeS,
        Long totalDistanceKm,
        LocalDateTime logDate
) {
    public static TourLogResponse from(TourLog log) {
        return new TourLogResponse(
                log.getTourLogId(),
                log.getTour().getTourId(),
                log.getComment(),
                log.getDifficulty(),
                log.getRating(),
                log.getTotalTimeS(),
                log.getTotalDistanceKm(),
                log.getLogDate()
        );
    }
}

File: frontend/src/app/features/tours/models/log/tour-log-response.ts
export interface TourLogResponse {
    tour_log_id: number;
    tour_id: number;
    comment: string;
    difficulty: number;
    rating: number;
    total_time_s: number; // in s
    total_distance_km: number; // in km
    log_date: string;
    is_owner: boolean;
}